flagscale/train/arguments.py

```diff
@@ -0,0 +1,228 @@
+
+
+import torch
+import types
+import ast
+import itertools
+from datetime import timedelta
+
+import torch
+
+from flagscale.train.hetero.parallel_context import RankMapper
+
+
+class FSTrainArguments:
+    """Extend the Megatron arguments with FlagScale specific arguments.
+    """
+    
+    def __init__(self, args, rank_mapper=None):
+        self.args = args
+        self._rank_mapper = rank_mapper
+
+    def __getattr__(self, name):
+        if name == "rank_mapper":
+            return self._rank_mapper
+        return getattr(self.args, name)
+
+    def __str__(self):
+        return str(self.args)
+
+    def __repr__(self):
+        return repr(self.args)
+    
+    def _initialize_distributed(self):
```

Comment:
Why do we need to move this distributed initialization here?

Reply:
To create the map between `logical_rank` and `physical rank`.